cpp/ql/src/semmle/code/cpp/Function.qll

```diff
@@ -1100,11 +1258,17 @@ class MoveAssignmentOperator extends Operator {
 
 
 /**
- * A C++ function which has a non-empty template argument list.
+ * A C++ function which has a non-empty template argument list. For example
+ * the function `myTemplateFunction` in the following code:
+ * ```
+ * template<class T>
+ * void myTemplateFunction(T t) {
+ *   ...
+ * }
+ * ```
  *
  * This includes function declarations which are immediately preceded by
- * `template <...>`, where the "..." part is not empty, and therefore it
- * does not include:
+ * `template <...>`, where the "..." part is not empty, but does not include:
```

Comment:
Why this change? Do you think one of the bullets below is not a logical consequence of this sentence?

If this sentence is indeed a precise description of `TemplateFunction`, then we can change "includes" to "comprises" to clarify that there are no other `TemplateFunction`s than those described here.

Reply:
Yes, I had a problem with "therefore" because you can't deduce that things aren't included from a statement that appears to be describing only some of the things that are.

I think it's supposed to describe all of them.  Changed to "comprises" and back to "therefore".